master/pkg/tasks/task_gc.go

```diff
@@ -37,10 +37,16 @@ func (g GCCkptSpec) ToTaskSpec() TaskSpec {
 	// Set Environment.
 	// Keep only the EnvironmentVariables provided by the experiment's config.
 	envVars := g.LegacyConfig.Environment.EnvironmentVariables()
+
+	podSpec := g.LegacyConfig.Environment.PodSpec()
+	if g.Base.TaskContainerDefaults.CheckpointGCPodSpec != nil {
```

Comment:
just to be super safe here, I don't think this can be nil but inside ```MergeIntoExpConfig``` there is a nil check for it.


```suggestion
	if g.Base.TaskContainerDefaults != nil && g.Base.TaskContainerDefaults.CheckpointGCPodSpec != nil {
```

Reply:
As `g.Base.TaskContainerDefaults` is of type mismatch with `nil`, it will be better to create a new func to check if it is `nil`. Something like this:

> `if !g.Base.TaskContainerDefaults.IsNil() && g.Base.TaskContainerDefaults.CheckpointGCPodSpec != nil `

```
func (c *TaskContainerDefaultsConfig) IsNil() bool {  
	if c != nil {
		return false
	}
	return true
}
```

Other option we have is to leverage the `validate` function and check for `err != nil`. Downside will be that `err` might not be empty because of other parameters being checked within that function. 

```
err := g.Base.TaskContainerDefaults.Validate()
if err != nil && g.Base.TaskContainerDefaults.CheckpointGCPodSpec != nil {
```

Let me know what do you think

Reply:
Oh my bad since ```g.Base.TaskContainersDefaultConfig``` is not a pointer we don't need to do anything

The code was good as is. I made a mistake thinking it was a pointer.

The ```IsNil``` check won't add anything since it will always not be nil.